newIDE/app/public/external/utils/path-editor.js

```diff
@@ -72,13 +81,59 @@ export const createPathEditorHeader = ({
     onSaveToGd(headerObject);
   });
   headerObject.cancelButton.addEventListener('click', onCancelChanges);
-  headerObject.saveFolderLabel.addEventListener('click', () => {
-    selectBaseFolderPath(headerObject);
-  });
-  headerObject.setFolderButton.addEventListener('click', () => {
+  const selectFolderPath = () => {
     selectBaseFolderPath(headerObject);
-  });
+  };
+  headerObject.saveFolderLabel.addEventListener('click', selectFolderPath);
+  headerObject.setFolderButton.addEventListener('click', selectFolderPath);
+
+  const openFolderPath = () => {
+    electron.shell.openItem(headerObject.state.folderPath)
+  }
+  headerObject.openFolderButton.addEventListener('click', openFolderPath);
+
+  /**
+ * Disables the path editor
+ */
+  headerObject.disableSavePathControls = () => {
+    headerObject.saveFolderLabel.removeEventListener('click', selectFolderPath);
+    headerObject.nameInput.style.color = '#8bb0b2';
+    headerObject.nameInput.style.border = '2px solid black';
+    headerObject.nameInput.disabled = true;
+    headerObject.saveFolderLabel.style.color = '#8bb0b2';
+    headerObject.saveFolderLabel.title =
+      'Changing the path is disabled on imported GD animations!';
+    headerObject.setFolderButton.removeEventListener('click', selectFolderPath);
+    headerObject.setFolderButton.style.visibility = 'hidden';
+  };
+
+  /**
+ * Returns a path for a file that does not exist yet.
+ * Used to avoid unwanted file overwriting.
+ */
+  headerObject.makeFileNameUnique = (filePath, missingExtension) => {
+    if (!fileExists(filePath)) {
+      return filePath;
+    }
+    const folderPath = path.dirname(filePath);
+    let extension = path.extname(filePath);
```

Comment:
nitpicking: you can write const `extension = path.extname(filePath) || missingExtension` instead of 

```
let extension = path.extname(filePath);
if (!extension) {
  extension = missingExtension;
}
```